rust/src/tests/errors/erofs.rs

```diff
@@ -0,0 +1,182 @@
+use std::{
+    panic::{catch_unwind, resume_unwind, AssertUnwindSafe},
+    path::Path,
+    process::Command,
+};
+
+use crate::utils::get_mountpoint;
+
+enum RemountOptions {
+    ReadOnly,
+    ReadWrite,
+}
+
+/// Guard to allow execution of this test only if it's allowed to run.
+pub(crate) fn can_run_erofs(conf: &crate::config::Config, _: &Path) -> anyhow::Result<()> {
+    if !conf.settings.allow_erofs {
+        anyhow::bail!("EROFS is not enabled in the configuration file")
+    }
+
+    Ok(())
+}
+
+/// Remount the file system mounted at `mountpoint` with the provided options.
+fn remount<P: AsRef<Path>>(mountpoint: P, options: RemountOptions) -> Result<(), anyhow::Error> {
+    if mountpoint.as_ref().parent().is_none() {
+        anyhow::bail!("Cannot remount root file system")
+    }
+
+    let opt = match options {
+        RemountOptions::ReadOnly => "ro",
+        RemountOptions::ReadWrite => "rw",
+    };
+
+    #[cfg(target_os = "linux")]
+    let opt = format!("remount,{opt}");
+
+    let mut cmd = Command::new("mount");
+
+    #[cfg(any(
+        target_os = "freebsd",
+        target_os = "netbsd",
+        target_os = "openbsd",
+        target_os = "dragonfly"
+    ))]
+    cmd.arg("-u");
+
+    // XXX: It is possible to do it directly with mount(2) but we use the CLI for now
```

Comment:
Pro tip: it's basically impossible to use mount(2) on FreeBSD.  You should use nmount(2) instead.

Reply:
Hmm, it doesn't look much harder to implement than mount(2). Should I implement it before merging?

Reply:
Sure, if you think you can do it easily.  Nix already has bindings for nmount.